api/src/contracttest/java/org/pmiops/workbench/consumer/DisksApiTest.java

```diff
@@ -0,0 +1,216 @@
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import au.com.dius.pact.consumer.MockServer;
+import au.com.dius.pact.consumer.dsl.PactDslWithProvider;
+import au.com.dius.pact.consumer.junit5.PactConsumerTestExt;
+import au.com.dius.pact.consumer.junit5.PactTestFor;
+import au.com.dius.pact.core.model.RequestResponsePact;
+import au.com.dius.pact.core.model.annotations.Pact;
+import io.pactfoundation.consumer.dsl.LambdaDslObject;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.pmiops.workbench.leonardo.ApiClient;
+import org.pmiops.workbench.leonardo.ApiException;
+import org.pmiops.workbench.leonardo.api.DisksApi;
+import org.pmiops.workbench.leonardo.model.LeonardoDiskType;
+import org.pmiops.workbench.leonardo.model.LeonardoUpdateDiskRequest;
+
+@ExtendWith(PactConsumerTestExt.class)
+class DisksApiTest {
+  private static final String GOOGLE_PROJECT = "googleproject";
+  private static final String DISK_NAME = "diskname";
+
+  private static final String DISK_ENDPOINT =
+      String.format("/api/google/v1/disks/%s/%s", GOOGLE_PROJECT, DISK_NAME);
+  private static final String LIST_DISKS_ENDPOINT =
+      String.format("/api/google/v1/disks/%s", GOOGLE_PROJECT);
+
+  void applyDiskExpectations(LambdaDslObject disk) {
+    disk.numberType("id", 3);
+    disk.object(
+        "cloudContext",
+        cloudContext -> {
+          cloudContext.stringType("cloudProvider");
+          cloudContext.stringType("cloudResource");
+        });
+    disk.stringType("zone");
+    disk.stringType("name", DISK_NAME);
+    disk.stringMatcher("status", "Ready|Deleting|Error", "Ready");
+    disk.object(
+        "auditInfo",
+        auditInfo -> {
+          auditInfo.stringType("creator");
+          auditInfo.stringType("createdDate", "2021-01-01T00:00:00Z");
+          auditInfo.stringType("dateAccessed", "2021-01-01T00:00:00Z");
+        });
+    disk.numberType("size");
+    disk.stringMatcher("diskType", "pd-standard|pd-ssd|pd-balanced", "pd-ssd");
+    disk.numberType("blockSize");
+    disk.object("labels", labels -> {});
+  }
+
+  @Pact(consumer = "aou-rwb-api", provider = "leonardo")
+  RequestResponsePact getDisk(PactDslWithProvider builder) {
+    return builder
+        .given("there is a disk in a Google project")
+        .uponReceiving("a request to get a disk")
+        .method("GET")
+        .path(DISK_ENDPOINT)
+        .headers(contentTypeJsonHeader)
+        .willRespondWith()
+        .status(200)
+        .headers(contentTypeJsonHeader)
+        .body(newJsonBody(this::applyDiskExpectations).build())
+        .toPact();
+  }
+
+  @Test
+  @PactTestFor(pactMethod = "getDisk")
+  void testGetDiskWhenDiskExists(MockServer mockServer) throws ApiException {
```

Comment:
my IntelliJ doesn't think any of the "throws ApiException" in this file are necessary

Reply:
Good catch. Thanks!